File: management-cluster-bootstrap/src/test/scala/doc/org/apache/pekko/management/cluster/bootstrap/ClusterBootstrapCompileOnly.scala
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, which was derived from Akka.
 */

/*
 * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package doc.org.apache.pekko.management.cluster.bootstrap

import org.apache.pekko
import pekko.actor.ActorSystem
import pekko.management.cluster.bootstrap.ClusterBootstrap
import pekko.management.scaladsl.PekkoManagement

object ClusterBootstrapCompileOnly {

  val system: ActorSystem = ActorSystem()

  // #start
  // Pekko Management hosts the HTTP routes used by bootstrap
  PekkoManagement(system).start()

  // Starting the bootstrap process needs to be done explicitly
  ClusterBootstrap(system).start()
  // #start

}

File: management-cluster-bootstrap/src/test/java/jdoc/org/apache/pekko/management/cluster/bootstrap/ClusterBootstrapCompileOnly.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * license agreements; and to You under the Apache License, version 2.0:
 *
 *   https://www.apache.org/licenses/LICENSE-2.0
 *
 * This file is part of the Apache Pekko project, derived from Akka.
 */

/*
 * Copyright (C) 2017-2021 Lightbend Inc. <https://www.lightbend.com>
 */

package jdoc.org.apache.pekko.management.cluster.bootstrap;

import org.apache.pekko.actor.ActorSystem;
import org.apache.pekko.management.scaladsl.PekkoManagement;
import org.apache.pekko.management.cluster.bootstrap.ClusterBootstrap;

public class ClusterBootstrapCompileOnly {
  public static void bootstrap() {

    ActorSystem system = ActorSystem.create();

    // #start
    // Pekko Management hosts the HTTP routes used by bootstrap
    PekkoManagement.get(system).start();

    // Starting the bootstrap process needs to be done explicitly
    ClusterBootstrap.get(system).start();
    // #start
  }
}
